### PR TITLE
feat: add Polar.sh provider

### LINKED ISSUE #65: Polar.sh support
### Clear and concise description of the problem

Add a support for [Polar.sh](https://polar.sh/) provider to sponsorkit

### Suggested solution

I can work on a PR if you're OK with adding it to `sponsorkit` ! 

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by Julien-R44**
### Description

Add [Polar](https://polar.sh/) as provider.

### Linked Issues

Close #65 

### Additional context

There's some information I don't have right now because i am new to Polar.sh. For example : I'm not sure what the `status` of a subscriptions means. Does an `inactive` status mean that the person is a past sponsor? 

I'll soon have answers to these questions, when some people will unsubscribe from my Polar. And I will do a new PR to update the code if there's anything wrong with our implementation. In the meantime, we're just keeping the `active` subscriptions and those with a defined `price`. and it seems to working fine

### CODE REVIEW

File: src/providers/polar.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    subscriptions.push(...subs.items)
+
+    pages = subs.pagination.max_page
+    page += 1
+  } while (page <= pages)
+
+  return subscriptions
+    /**
+     * - People can subscribe for free on Polar : the price is null in this case
+     * - We also only keep `active` subscriptions. Still not sure what `inactive` means
```
Comment (antfu) [inline src/providers/polar.ts]: Active probably means past sponsors (monthly sponsors than canceled at some point)?

We do support listing past sponsors, where `monthlyDollars` should be `-1` for them and up to renderers to render them or not

Comment (Julien-R44) [reply src/providers/polar.ts]: Yeah probably, that's what I thought too. But I'm not sure that the returned object will retains the same shape if `inactive`. That's why I suggested keeping the code as it is, until I or someone else can test the call API with past sponsorship, and adapt the code accordingly

But let's try it like that. I updated the code and and will keep an eye out as soon as anyone unsubscribes from my polar 👀

File: src/types.ts [hunk 2]
```diff
@@ -155,6 +155,15 @@ export interface ProvidersConfig {
      */
     exechangeRate?: number
   }
+
+  polar?: {
+    /**
+     * Polar token that have access to your sponsorships.
+     *
+     * Will read from `SPONSORKIT_POLAR_TOKEN` environment variable if not set.
```
Comment (antfu) [inline src/types.ts:163]:
```suggestion
     * Will read from `SPONSORKIT_POLAR_TOKEN` environment variable if not set.
     *
     * @see https://polar.sh/settings
     * @deprecated It's not recommended set this value directly, pass from env or use `.env` file.
```

**Review by antfu — APPROVED** [review]
Thanks for the clean and prompt implementation!

**Comment by birkjernstrom** [comment]
Oh wow - thank you so much @Julien-R44 for adding this & @antfu for building this great kit ♥️

**Pull request merged**